server/src/main/java/org/elasticsearch/readiness/ReadinessService.java

```diff
@@ -102,18 +107,32 @@ InetSocketAddress socketAddress(InetAddress host, int portNumber) {
 
     // package private for testing
     ServerSocketChannel setupSocket() {
-        InetAddress localhost = InetAddress.getLoopbackAddress();
-        int portNumber = PORT.get(environment.settings());
+        var settings = environment.settings();
+        int portNumber = PORT.get(settings);
         assert portNumber >= 0;
 
+        List<String> httpBindHost = SETTING_HTTP_BIND_HOST.get(settings);
+        var bindHosts = (httpBindHost.isEmpty() ? NetworkService.GLOBAL_NETWORK_BIND_HOST_SETTING.get(settings) : httpBindHost).toArray(
+            Strings.EMPTY_ARRAY
```

Comment:
There seems to be two levels of configuration before the fallback to the loopback.   Do we need the configuration checks? And should the fallback be to the wildcard address which implies *all* (rather than to the loopback) ?

Reply:
I struggled with this a lot, what's the right address to pick, 0, publish, bound? I borrowed this code from how we  setup the bound addresses in `AbstractHttpServerTransport`, I don't mind binding to 0 (all) as fallback, but if a user doesn't configure any of those above for Elasticsearch, they will get ES bound to localhost for HTTP, unless they configure `SETTING_HTTP_PUBLISH_HOST`. I don't know if binding to localhost and publishing another host address is a valid configuration?

Reply:
I discussed it with @rjernst just now, we decided that perhaps the easiest is to bind for 0 and if there are any concerns firewall rules should be used to limit access to the port.